### PR TITLE
feat(frontend): support `search_path` and improve support for "schema-qualified name"

### LINKED ISSUE #5447: feat(frontend): support search path for binder and other functions like current_schemas
### Is your feature request related to a problem? Please describe.

Currently when binding any relations in frontend, binder will only search under `public` schema if user don't specified. But for postgreSQL, `pg_catalog` will be also took into consideration and user can also set search path in session as follows:

```sql
CREATE SCHEMA collate_tests;
SET search_path = collate_tests;
```

Besides, to integrate with other tools like pgcli, which will send some internal queries that don't specify `pg_catalog`, we might need to support search path in `Binder`.

Cc @xiangjinwu @HuaHuaY 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #5723: bug(frontend): panic when create index in a schema different from the schema in which the table is located
### Describe the bug

Frontend will panic when we create index in a different schema from the schema in which the table is located. In PostgreSQL, no schema name should be included in index's name and index will be in the same schema of the table. In our database, frontend will search the table in the schema of index, so it cann't find the table and `unwrap()` will panic.
We plan to follow PostgreSQL's behavior and ban `create index xxx.yyy on ...`.

### To Reproduce

```sql
dev=> create table tmp(id int);
CREATE_TABLE
dev=> create schema tmp;
CREATE_SCHEMA
dev=> create index tmp.tmp1 on tmp(id);
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!?>
```
frontend-4566.log
```
thread 'risingwave-main' panicked at 'called `Option::unwrap()` on a `None` value', src/frontend/src/catalog/schema_catalog.rs:89:14
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:48:5
   3: core::option::Option<T>::unwrap
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/option.rs:775:21
   4: risingwave_frontend::catalog::schema_catalog::SchemaCatalog::create_index
             at ./src/frontend/src/catalog/schema_catalog.rs:87:29
```

### Expected behavior

```sql
postgres=# create table tmp(id int);
CREATE TABLE
postgres=# create schema tmp;
CREATE SCHEMA
postgres=# create index tmp.tmp1 on tmp(id);
ERROR:  syntax error at or near "."
LINE 1: create index tmp.tmp1 on tmp(id);
                        ^
postgres=#
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HuaHuaY**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

There are two major changes:
1. support `search_path` session variable (PostgreSQL's doc: https://www.postgresql.org/docs/14/runtime-config-client.html#GUC-SEARCH-PATH). 
    Its value is a vector of `String`. Its default value is `"$user", public`. **When an item's value is "$user"(include double quotes), it will be replaced with the user's name of the session when it's being used.**
    - when we `select` or `drop` object and don't give a specified schema, it will search the object from the valid items in schema `pg_catalog` and `search_path`. **If schema `pg_catalog` is not in `search_path`, we will search `pg_catalog` first.** If schema `pg_catalog` is in `search_path`, we will follow the order in `search_path`.
    - when we `create` a `source` or `mv` and don't give a specified schema, it will use the first valid schema in `search_path`.
    - when we `create` a `index` or `sink`, it will use the schema of the associated table.
2. improve support for "schema-qualified name" (https://www.postgresql.org/docs/current/ddl-schemas.html).
    - We can use "database.schema.table" or "database.schema.source" when referring to a table or a source. **"database" must be the same as the database of the session.** We can also use "schema.table" or "schema.source" when referring to a table or a source.
    - We can use "database.schema.index" or "database.schema.sink" when dropping an index or a sink. **"database" must be the same as the database of the session.** We can also use "schema.index" or "schema.sink" when dropping an index or a sink.
    - We can only use name without schema when creating an index or a sink.
    - We can only use name without database when referring to a schema.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators

### Release note

1. support `search_path`(PostgreSQL's doc: https://www.postgresql.org/docs/14/runtime-config-client.html#GUC-SEARCH-PATH). 
    This variable specifies the order in which schemas are searched when an object (table, source, etc.) is referenced by a simple name with no schema specified. When there are objects of identical names in different schemas, the one found first in the search path is used. An object that is not in any of the schemas in the search path can only be referenced by specifying its containing schema with a qualified (dotted) name.

## Refer to a related PR or issue link (optional)
close #5723
close #5447

### CODE REVIEW

File: src/frontend/src/binder/relation/table_or_source.rs [hunk 1]
```diff
@@ -114,15 +101,93 @@ impl Binder {
...
+                    let search_path_left =
+                        SearchPath::from_str(&paths[0..pg_catalog_position].join(", "))?;
+                    let schema_path_left = SchemaPath::Path(&search_path_left, user_name);
+                    let search_path_right = SearchPath::from_str(
+                        &paths[pg_catalog_position + 1..paths.len()].join(", "),
+                    )?;
+                    let schema_path_right = SchemaPath::Path(&search_path_right, user_name);
+                    let schema_path = SchemaPath::Path(&self.search_path, user_name);
+
+                    if let Ok((table_catalog, schema_name)) =
```
Comment (yezizp2012) [inline src/frontend/src/binder/relation/table_or_source.rs:168]: I tested in postgreSQL, system tables have a higher priority when tables with name  like `pg_xxxx` existed in both `pg_catalog` and other schemas in search path:
```
postgres=# create table xxx.pg_index(v int);
CREATE TABLE
postgres=# set search_path to xxx;
SET
postgres=# select * from pg_index;
/// show data in pg_catalog.pg_index
postgres=# select * from xxx.pg_index;
 v
---
(0 rows)
postgres=# \dt pg_index
           List of relations
   Schema   |   Name   | Type  | Owner
------------+----------+-------+--------
 pg_catalog | pg_index | table | august
(1 row)

postgres=# \dt xxx.pg_index
         List of relations
 Schema |   Name   | Type  | Owner
--------+----------+-------+--------
 xxx    | pg_index | table | august
(1 row)
```

And luckily in implementation of this PR left part of search path will be alway empty as I comment in L160. How about simplify these codes as follows:
1. resolve system tables, return if success.
2. resolve tables and sources on search_path.real_path(), return if success.
3. not found.
if so, maybe there's no need to keep hidden `pg_catalog` in search_path IIUC.

**Review by yezizp2012 — COMMENTED** [review]
Good job!! The rest LGTM. Please also add some release note because `set search_path to xxxx` is a good feature and should be user-facing.

File: src/frontend/src/binder/relation/table_or_source.rs (hunk 1 above)
Comment (HuaHuaY) [reply src/frontend/src/binder/relation/table_or_source.rs:168]: If you set search_path to `xxx, pg_catalog`, you will get a different answer. So I divide it to two parts.

Comment (yezizp2012) [reply src/frontend/src/binder/relation/table_or_source.rs:168]: Got it, let's simplify these code as we discussed offline as follows:
```rust
for path in search_path.path() {
    if path == PG_CATALOG_SCHEMA_NAME {
        get_sys_table_by_name(xxx, yyy)
    } else {
        get_table_by_name(xxx, SchemaPath::Name(path), yyy)
    }
}
```

**Review by yezizp2012 — APPROVED** [review]
LGTM~

**Pull request merged**